pyERGM/metrics.py

```diff
@@ -1169,3 +1172,27 @@ def get_ignored_features(self):
             parameter_names += metric._get_ignored_features()
 
         return parameter_names
+
+
+class SumDistancesConnectedNeurons(Metric):
+    """
+    This function calculates the sum of euclidean distances between all pairs of connected neurons.
+    rows = samples, columns = axes
+    """
+
+    def __init__(self, positions):
+        super().__init__(requires_graph=False)
+        self._is_directed = True
+        self._is_dyadic_independent = True
+        self._positions = positions
+
+    def calculate(self, input_graph: np.ndarray):
+        if isinstance(self._positions, (pd.DataFrame, pd.Series)):
+            self._positions = self._positions.values
+        if len(self._positions.shape) == 1:
+            self._positions = self._positions.reshape(-1, 1)
+        dist_matrix = distance.pdist(self._positions, metric='euclidean')
+        dist_square = distance.squareform(dist_matrix)
+        distances = dist_square[np.where(input_graph)]
+        return sum(distances)
```

Comment:
The calculation of the sum of the distances can more efficient in terms of runtime. Let's discuss that

Reply:
done as discussed